pkg/arvo/app/invite-hook.hoon

```diff
@@ -1,123 +1,123 @@
-::  invite-hook [landscape]:
+::  invite-hook [landscape]: receive invites from any source
 ::
-::  receive invites from any source
+::    only handles %invite actions:
+::    - can be poked by the host team to send an invite out to someone.
+::    - can be poked by foreign ships to send an invite to us.
 ::
-::    only handles %invite actions. accepts json, but only from the host team.
-::    can be poked by the host team to send an invite out to someone.
-::    can be poked by foreign ships to send an invite to us.
-::
-/+  *invite-json, default-agent, verb, dbug
+/-  *invite-store
+/+  default-agent, dbug
 ::
 |%
 +$  state-0  [%0 ~]
-::
 +$  card  card:agent:gall
 --
 ::
 =|  state-0
 =*  state  -
-::
-%+  verb  |
 %-  agent:dbug
 ^-  agent:gall
-=<
-  |_  =bowl:gall
-  +*  this  .
-      do    ~(. +> bowl)
-      def   ~(. (default-agent this %|) bowl)
-  ::
-  ++  on-init
-    ^-  (quip card _this)
-    [~ this]
-  ::
-  ++  on-save  !>(state)
-  ++  on-load
-    |=  old=vase
-    ^-  (quip card _this)
-    [~ this(state !<(state-0 old))]
-  ::
-  ++  on-poke
-    |=  [=mark =vase]
-    ^-  (quip card _this)
-    :_  this
-    ?+  mark  (on-poke:def mark vase)
-        %json
-      ::  only accept json from ourselves.
+::
+|_  =bowl:gall
++*  this  .
+    def   ~(. (default-agent this %|) bowl)
+::
+++  on-init  [~ this]
+++  on-save  !>(state)
+++  on-load
+  |=  old=vase
+  ^-  (quip card _this)
+  [~ this(state !<(state-0 old))]
+::
+++  on-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  |^
+  :_  this
+  ?+  mark  (on-poke:def mark vase)
+      %invite-action
+    =/  act=action  !<(action vase)
+    ?+  -.act     ~
+        %invites
+      ?.  (team:title [our src]:bowl)  ~
+      ::  outgoing. we must be inviting other ships. send them each an invite
       ::
-      ?>  (team:title our.bowl src.bowl)
-      =/  act  (json-to-action !<(json vase))
-      ?>  ?=(%invite -.act)
-      [(invite-hook-poke:do recipient.invite.act act)]~
+      %+  turn  ~(tap in recipients.invites.act)
+      |=  recipient=ship
+      ^-  card
+      ?<  (team:title our.bowl recipient)
+      %+  invite-hook-poke  recipient
+      :^  %invite 
```

Comment:
`%invite` should be indented once more -- or the next two args could be lifted onto this line, either way.